core/embed/rust/src/zcash/orchard_bridge.rs

```diff
@@ -0,0 +1,94 @@
+use core::convert::{TryFrom, TryInto};
+use core::ops::{Deref, DerefMut};
+
+use cstr_core::CStr;
+use f4jumble;
+use orchard::keys::{FullViewingKey, IncomingViewingKey, SpendingKey};
+
+use crate::error::Error;
+use crate::micropython::{
+    buffer::{Buffer, BufferMut},
+    obj::Obj,
+};
+use crate::util;
+
+/// Returns Orchard Full Viewing Key.
+fn get_orchard_fvk(sk: Obj) -> Result<FullViewingKey, Error> {
+    let sk: Buffer = sk.try_into()?;
+    let mut sk_bytes = [0u8; 32];
+    sk_bytes.copy_from_slice(sk.deref());
```

Comment:
I think you can just use `&sk` here instead of explicitly `deref()`ing.

Reply:
Also, and this would be an actual problem: `copy_from_slice` will panic in case `sk` has a `len() != 32`. I suppose we can do `let sk_bytes = sk.try_into()?`, or maybe `sk.as_ref().try_into()?` if the auto-deref doesn't kick in in this case.

Reply:
Fixed.

Reply:
Fixed.